addons/box/apps.py

```diff
@@ -1,9 +1,95 @@
+import requests
 from addons.base.apps import BaseAddonAppConfig, generic_root_folder
 from addons.box.settings import MAX_UPLOAD_SIZE
-
+from website.settings import GV_USER_DOMAIN, GV_RESOURCE_DOMAIN
 
 box_root_folder = generic_root_folder('box')
 
+
+class GravyValetBoxAddonAppConfig():
+    class MockNodeSetting:
+        def __init__(self, resource, auth):
+            resp = requests.get(
+                GV_RESOURCE_DOMAIN.format(owner_uri=resource.absolute_url),
+                auth=(auth.user.username, auth.user.password)
+            )
+            from addons.box.models import NodeSettings
+            settings_obj, created = NodeSettings.objects.get_or_create(owner=resource)
+
+            # No addon found in GV
+            if resp.status_code == 404 and created:
+                resp.raise_for_status()
+
+            # addon disabled on GV, but not here
+            if resp.status_code == 404 or resp.status_code == 410:
+                settings_obj.delete()
+                resp.raise_for_status()
```

Comment:
Let's just ignore the case where something hasn't been migrated yet for now. :ostrich: